geniza/corpus/annotation_export.py

```diff
@@ -0,0 +1,215 @@
+import json
+import os
+from collections import defaultdict
+from urllib.parse import urlparse
+
+from django.conf import settings
+from django.core.management.base import BaseCommand, CommandError
+from django.template.defaultfilters import pluralize
+from django.template.loader import get_template
+from django.utils.text import slugify
+from git import InvalidGitRepositoryError, Repo
+
+from geniza.annotations.models import Annotation, annotations_to_list
+from geniza.corpus.models import Document
+from geniza.footnotes.models import Footnote
+
+
+class AnnotationExporter:
+    def __init__(self, pgpids=None, stdout=None, push_changes=True):
+        if not getattr(settings, "ANNOTATION_BACKUP_PATH") or not getattr(
+            settings, "ANNOTATION_BACKUP_GITREPO"
+        ):
+            raise Exception(
+                "Settings for ANNOTATION_BACKUP_PATH and ANNOTATION_BACKUP_GITREPO are required"
+            )
+
+        base_output_dir = settings.ANNOTATION_BACKUP_PATH
+        # initialize git repo interface for output path
+        if not push_changes:
+            # no sync or cloning
+            repo = Repo(base_output_dir)
+        else:
+            repo = self.setup_repo(base_output_dir, settings.ANNOTATION_BACKUP_GITREPO)
+        annotations_output_dir = os.path.join(base_output_dir, "annotations")
+
+        # define paths and ensure directories exist for compiled transcription
+        transcription_output_dir = {}
+        for output_format in ["txt", "html"]:
+            format_path = os.path.join(base_output_dir, "transcriptions", output_format)
+            transcription_output_dir[output_format] = format_path
+            os.makedirs(format_path, exist_ok=True)
+
+        # identify content to backup based on documents with digital editions
+
+        docs = Document.objects.filter(
+            footnotes__doc_relation__contains=Footnote.DIGITAL_EDITION
+        ).distinct()
+        # if ids are specified, limit to just those documents
+        if pgpids:
+            docs = docs.filter(pk__in=pgpids)
+        # use logging? and/or pass in stdout when running from command?
```

Comment:
Maybe rename param something like "logger" and then pass stdout when running from command, and use django logger as default.

Reply:
that would be a nice solution, but they don't have the same interface...